### PR TITLE
apple-fonts: init at 1.0.3

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes
initialized apple-fonts

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
Please move the package to `pkgs/by-name/ap/apple-fonts/package.nix`, thanks.

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:25]:
```suggestion
let
  pro = fetchurl {
    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Pro.dmg";
    hash = "sha256-nkuHge3/Vy8lwYx9z+pvsQZfzrNIP4K0OutpPl4yXn0=";
  };

  compact = fetchurl {
    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Compact.dmg";
    hash = "sha256-+Q4HInJBl3FLb29/x9utf7A55uh5r79eh/7hdQDdbSI=";
  };

  mono = fetchurl {
    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Mono.dmg";
    hash = "sha256-pqkYgJZttKKHqTYobBUjud0fW79dS5tdzYJ23we9TW4=";
  };

  ny = fetchurl {
    url = "https://devimages-cdn.apple.com/design/resources/download/NY.dmg";
    hash = "sha256-XOiWc4c7Yah+mM7axk8g1gY12vXamQF78Keqd3/0/cE=";
  };
in
stdenv.mkDerivation rec {
  pname = "apple-fonts";
  version = "1.0.3";

```

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:5]: Where does this version come from?

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:3]: Use `stdenvNoCC` if no build is required.

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:33]:
```suggestion
  installPhase = ''
    runHook preInstall
```

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:62]:
```suggestion
    rm -rf $out/fontfiles

    runHook postInstall
  '';
```

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:68]: Please add yourself as a maintainer in a separate commit before adding the package.

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (wegank) [inline pkgs/data/fonts/apple-fonts/default.nix:64]: `meta.platforms` is missing.

**Comment by wegank** [comment]
> This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:
> 
> https://discourse.nixos.org/t/prs-ready-for-review/3032/3235

It's too early to declare the PR as ready, especially if CI hasn't even finished evaluating and building the package.

File: pkgs/data/fonts/apple-fonts/default.nix
Comment (TheDuckWithAMonocle) [reply pkgs/data/fonts/apple-fonts/default.nix:5]: it's coming from the AUR package they have maintained, wasn't sure how this would be, so i just kept the same value

**Comment by TheDuckWithAMonocle** [comment]
Sorry about that, I'm still very new to this. I'll make the changes and do another commit

**Review by adamcstephens — CHANGES_REQUESTED** [review]
These URLs can and do change, so these packages are not reproducible. Unless you can find a stable source for the fonts, we should not merge this. 

I know this because I’ve had these URLs change on me. https://codeberg.org/adamcstephens/apple-fonts.nix/commit/538f773302264917ea0b5acef475f3345daf1315

**Comment by TheDuckWithAMonocle** [comment]
it could be possible to upload these dmg files to internet archive and fetch them from there. 
there seems to be a package of the [monaco typeface](https://archive.org/details/Monaco_13) which is still up

**Comment by Luflosi** [comment]
Please add another commit before the others, where you add yourself to the list of maintainers. You can take ff80331317ac50b80c8cb15918421885d341d63e as inspiration.
Please also squash the three commits you have now into one and fix remove all the duplicate lines from the commit message.

**Pull request closed without merging**
